### PR TITLE
Comet integration updates

### PR DESCRIPTION
**Pull request by DN6**
This PR:

Updates the Comet ML integration with YOLOv8 

This PR:

1. Adds `Comet from` source tracking to YOLOv8
2. Adds support for logging training and validation images and model checkpoints every `N` epochs during training
3. Adds support for logging interactive Confusion Matrix to Comet
4. Adds support for Online and Offline Experiment logging 
5. Updates  Comet UTM links in the main README. 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Updated integration with Comet in the YOLOv8 code base.

### 📊 Key Changes
- Updated Comet integration links and documentation for YOLOv8.
- Introduced new Comet logging features and environment variables for customization.
- Enhanced logging of model information, validation predictions, confusion matrix, and various plots to Comet during training.
- Implemented functions to adjust bounding box annotations to the original image scale.
- Added support for offline Comet experiments.
- Provided a structured way to handle multi-process limitations when creating Comet experiments.

### 🎯 Purpose & Impact
- **For Developers**: Helps track and visualize the model training process more effectively, especially with large-scale or distributed training setups.
- **For End-Users**: Provides clearer insights into model performance and improves the debugging process by visualizing predictions and metrics related to the trained models.
- Overall, the update leads to a more comprehensive and user-friendly model training experience with valuable insights for performance optimization. 🚀

### CODE REVIEW

**Comment by DN6** [comment]
I have read the CLA Document and I sign the CLA

**Comment by DN6** [comment]
@glenn-jocher @AyushExel Submitting an update for the Comet integration. Would appreciate it if I could get your review on it.

**Comment by DN6** [comment]
Hi @glenn-jocher. The update details are in the PR description. Would you like me to expand on the specifics?

**Comment by glenn-jocher** [comment]
@DN6 thank you for letting me know that the update details are in the PR description. I will take a look at it and review your changes. If I have any questions or need additional information, I'll be sure to let you know.

**Comment by glenn-jocher** [comment]
@DN6 ok, I've reviewed and cleaned it up a bit: re-introduced our standard warning when loggers aren't initialized, created a new bitly link that points to your UTM at https://bit.ly/yolov8-readme-comet and updated all 4 link locations in English and Chinese READMEs.

File: ultralytics/yolo/utils/callbacks/comet.py [hunk 1]
```diff
@@ -10,41 +13,319 @@
...
+def _create_prediction_metadata_map(model_predictions):
+    pred_metadata_map = {}
+    for prediction in model_predictions:
+        pred_metadata_map.setdefault(prediction['image_id'], [])
+        pred_metadata_map[prediction['image_id']].append(prediction)
+
+    return pred_metadata_map
+
+
+def _log_confusion_matrix(experiment, trainer, curr_step, curr_epoch):
```
Comment (glenn-jocher) [inline ultralytics/yolo/utils/callbacks/comet.py:172]: @DN6 unless I'm missing something here, this Confusion matrix will be incorrect as you are sorting class names alphabetically when the validator confusion matrix has been created with a class index sorting. This means you are detaching the class names from their corresponding index.

**Comment by glenn-jocher** [comment]
@DN6 I've got an apology in order here! I accidentally closed this when I merged 8.0.62 yesterday as I didn't realize this was targeted to `updates` branch rather than main. I'll go ahead and create a new PR since it doesn't seem possible to reopen.

**Pull request closed without merging**